### ISSUE TITLE
[TW-345] Show command highlight configuration

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#380]
_Nicolas Appriou on 2014-01-29T15:50:52Z says:_ 

 When using the show command, some lines are highlighted:
<pre>
Some of your .taskrc variables differ from the default values.  These are highlighted in color above.
</pre>
or
<pre>
Your .taskrc file contains these unrecognized variables:

  These are highlighted in color above.
</pre>

For the moment, the highlight color is hardcoded in src/commands/CmdShow.cpp . This color should be a configuration option.

I suggest using the config keys _color.warning_ and _color.error_ .

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-01-29T15:50:52Z
Modified: 2014-02-09T00:53:17Z
```

**Comment by taskwarrior** [comment]
_David Patrick on 2014-01-29T19:42:45Z says:_ 

 Good observation Nicolas, the "task show" color-rules are hard-coded, and maybe shouldn't be.

If that were configurable in.taskrc, like other colors, then it would be important to make it clear what each configuration does, and is "show" specific, maybe something like:

   color.show.non-default=rgb123
color.show.unrecognized=rgb321
color.show.deprecated=rgb234

Are there other elements of task show that could use color rules?

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-01-29T23:06:31Z says:_ 

 Thanks for reporting this.

**Comment by taskwarrior** [comment]
_Nicolas Appriou on 2014-01-30T06:12:34Z says:_ 

 I'm not sure that those color are "show specific". I'm using taskwarrior for less than a week so I may not have explored every functionnality.

Here is the incriminated code.
<pre>
// src/commands/CmdShow.cpp
Color error ("bold white on red");
Color warning ("black on yellow");
</pre>

The label for these colors are pretty generic. I think we should have some generic warning/error/other(?) color for message display.

I found another unconfigurable color:
<pre>
$ task list
There are different numbers of columns and labels for report 'list'
</pre>

This message is clearly an error but is displayed in another color (yellow) than the previously defined error message ("bold white on red").

This is why I suggested color.error and color.warning rather than color.show.(error|warning) in an attempt to globalize these settings.

I've patched task for the original issue, just need to run a few tests before releasing it. I don't now if it worth making it a global settings (lot of work, but seems to add a consistency in displayed message).

**Comment by taskwarrior** [comment]
_David Patrick on 2014-01-30T09:41:26Z says:_ 

 neat!

**Comment by taskwarrior** [comment]
_Nicolas Appriou on 2014-02-01T10:18:12Z says:_ 

 Here is a patch.
I didn't write unit test, I'm not used to C++ unit tests and really didn't know how to test this specific case.

I noticed that there was already an color.error configuration. I changed the color from yellow to red and added a color.warning that prints in yellow.

There was already color.footnote configuration that had the yellow default color. Didn't change this one.
It seemed that the only hardcoded color was in src/commands/CmdShow.cpp
